c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -3341,7 +3356,11 @@ class DeviceCachingAllocator {
         EventPool::Event event = std::move(e.first);
         Block* block = e.second;
 
-        cudaError_t err = C10_CUDA_ERROR_HANDLED(cudaEventQuery(*event));
+        cudaError_t err{};
+        {
+          at::cuda::CUDAStreamCaptureModeGuard g{cudaStreamCaptureModeRelaxed};
+          err = C10_CUDA_ERROR_HANDLED(cudaEventQuery(*event));
```

Comment:
what do you think this is achieving both during capture and during replay? During capture it's a no-op, all events are immediately completed because there is no time during capture, during replay it's just not in the graph. 
The practical effect of this is that memory used on a different stream is immeadiately marked as reusable. This is a gross correctness error. 
The fact that allocator is not touched during replay does not help - the problem here is that memory would be reused that should not have been reused, because proper syncs are not made.